src/Adapter/Product/Combination/CommandHandler/UpdateCombinationHandler.php

```diff
@@ -70,5 +90,40 @@ public function handle(UpdateCombinationCommand $command): void
             $updatableProperties,
             CannotUpdateCombinationException::FAILED_UPDATE_COMBINATION
         );
+
+        // Only update default if the property is set AND is true
+        if (true === $command->isDefault()) {
+            $this->defaultCombinationUpdater->setDefaultCombination(
+                $command->getCombinationId(),
+                // @todo: temporary hardcoded shop constraint. Needs to be required in command constructor.
+                ShopConstraint::shop((int) \Context::getContext()->shop->id)
```

Comment:
I think we should pass the context shop id to the command instead using \Context class in the Handler.

Reply:
Yeah I agree, but this is part of a refacto that is done in another PR here https://github.com/PrestaShop/PrestaShop/pull/30320

So depending on which one is merged first I'll fix this in this PR or in a future one, so for now I left the same todo comment as in the initial handler

Reply:
Ok